### PR TITLE
Fix multi errors

### PR DESCRIPTION
**Pull request by hash-an**
⁃ In the initialise function, explained the values in the arrays: _initialAddress, _initialValue 

 ⁃ Linter errors: CurrentEpoch, NextEpoch should start with a small case letters.

 ⁃ Turn _checkVaultInMaintenance, _checkVaultOutMaintenance to modifiers.

 ⁃ In withdraw function withdraw we have:
address caller = _msgSender();
if ((_owner != caller) || (_receiver != caller)) {
    revert Errors.CallerIsNotOwnerOrReceiver(caller, _owner, _receiver);
}
 The check (_owner != caller) is duplicated because the modifier whitelisted is used. Removing this will save gas.

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -130,14 +130,14 @@ module.exports = {
...
-        gasPriceApi: `https://api-optimistic.etherscan.io/api?module=proxy&action=eth_gasPrice&apikey=${process.env.OPTIMISM_API_KEY}`,
+        gasPriceApi: `https://api-optimistic.etherscan.io/api?module=proxy&action=eth_gasPrice&apikey=${process.env.ETHERSCAN_API_KEY}`,
         // gasPrice: 35
     },
     etherscan: {
         // Your API key for Etherscan
         // Obtain one at https://etherscan.io/
-        // apiKey: process.env.ETHERSCAN_API_KEY,
-        apiKey: process.env.OPTIMISM_API_KEY,
+        apiKey: process.env.ETHERSCAN_API_KEY,
```
Comment (carlgauss221) [inline hardhat.config.ts:139]: need to keep config of optimism, becuase we still try with this network a little bit more!!

File: src/CalculumVault.sol [hunk 2]
```diff
@@ -119,12 +119,36 @@ contract CalculumVault is
...
+        if (
+            (block.timestamp <
+                (getNextEpoch().sub(MAINTENANCE_PERIOD_PRE_START))) ||
+            (block.timestamp > (getNextEpoch()))
+        ) {
+            revert Errors.VaultOutMaintenance();
+        }
+        _;
+    }
+
```
Comment (carlgauss221) [inline src/CalculumVault.sol:144]: This change is not Efficient, in the ByteSize Code aspect, becuase i try in my side and add 236 KB, more to the Byte Size of the smart contract!!!, check is you agree roll-back

File: test/index.test.ts [hunk 3]
```diff
@@ -3858,7 +3858,7 @@ describe("Verification of Basic Value and Features", function () {
                 .format("dddd, MMMM Do YYYY, h:mm:ss a")
         );
         // Getting Current Epoch and Next Epoch
-        await Calculum.connect(deployer).CurrentEpoch();
+        await Calculum.connect(deployer).currentEpoch();
```
Comment (carlgauss221) [inline test/index.test.ts:3861]: All changes aboout changes from Pascal case to Camel Case, i totally agree!!!

File: hardhat.config.ts (hunk 1 above)
Comment (hash-an) [reply hardhat.config.ts:139]: Why can't we just use etherscan api key? @carlgauss221

File: src/CalculumVault.sol (hunk 2 above)
Comment (hash-an) [reply src/CalculumVault.sol:144]: If the ByteSize is a concern here, we maybe should turn `whitelisted` into a private function as well. What do you think? @carlgauss221

**Comment by carlgauss221** [comment]
close

**Pull request closed without merging**